File: Lab2/hailstoneSequence.cpp
#include <iostream>
using namespace std;
int main() {
    int n;
    cout << "Please enter a number greater than 1:";
    cin >> n;

    while (n <=1) {
        cout << "Please enter a number greater than 1:";
        cin >> n;
    }
    int steps=0;
    cout <<n;
    while (n!=1) {
        if (n%2==0) {
            n = n/2;
        }
        else {
            n = n*3+1;
        }
        cout << n <<" -> ";
        steps++;
    }
    cout << "Total Steps: " << steps << endl;
}

File: Lab2/hailstoneSequence.py
x = int(input("Please enter a positive integer greater than 1."))
step = 0
while x > 1:
    print(x,end=" ")
    while x != 1:
        if x % 2 == 0:
            x = x // 2
            print("->", x, end=" ")
            step = step + 1

        else:
            x = 3 * x + 1
            print("->", x,end=" ")
            step = step + 1

print("Total Steps: ", step)
